src/rgw/rgw_crypt.cc

```diff
@@ -359,25 +359,26 @@ class AES_256_CBC : public BlockCrypt {
                      const unsigned char (&key)[AES_256_KEYSIZE],
                      bool encrypt)
   {
-    static std::atomic<bool> failed_to_get_crypto(false);
-    CryptoAccelRef crypto_accel;
-    if (! failed_to_get_crypto.load())
```

Comment:
this pointer isn't static though, so it wouldn't extend the plugin lifetime past that of CephContext

my point above was that, if this plugin lifetime only becomes an issue because of changes you're making in #14498, then we should address the issue in that pr. mainly because I expect most of this plugin-related code to move from `rgw_crypt.cc` into `auth/Crypto.cc` (right?)